app/src/test/java/org/mozilla/tv/firefox/architecture/KillswitchLinearLayoutTest.kt

```diff
@@ -0,0 +1,124 @@
+package org.mozilla.tv.firefox.architecture
+
+import android.app.Activity
+import android.view.View
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.tv.firefox.components.locale.LocaleManager
+import org.robolectric.Robolectric
+import org.robolectric.RobolectricTestRunner
+import java.util.Locale
+
+@RunWith(RobolectricTestRunner::class)
+class KillswitchLinearLayoutTest {
+
+    private lateinit var killswitchLayout: KillswitchLinearLayout
+
+    @Before
+    fun setup() {
+        val activityController = Robolectric.buildActivity(Activity::class.java)
+        killswitchLayout = KillswitchLinearLayout(activityController.get())
+    }
+
+    @Test
+    fun `GIVEN requirements are not set WHEN visibility is set THEN visibility should always be gone`() {
+        // Don't set requirements
+        visibilityIsAlwaysGone()
+    }
+
+    @Test
+    fun `GIVEN all locales are allowed WHEN visibility is set THEN it should be applied`() {
+        killswitchLayout.setRequirements(true, KillswitchLocales.All)
+
+        visibilityIsApplied()
+    }
+
+    @Test
+    fun `GIVEN experiment is disabled WHEN visibility is set THEN visibility should always be gone`() {
+        killswitchLayout.setRequirements(false, KillswitchLocales.All)
+
+        visibilityIsAlwaysGone()
+    }
+
+    @Test
+    fun `GIVEN visibility is set WHEN requirements are set THEN visibility should match the earlier set values`() {
+        killswitchLayout.visibility = View.VISIBLE
+        assertEquals(View.GONE, killswitchLayout.visibility)
+
+        killswitchLayout.setRequirements(true, KillswitchLocales.All)
+        assertEquals(View.VISIBLE, killswitchLayout.visibility)
+
+        visibilityIsApplied()
```

Comment:
nit: I think it's good to test this last assertion too but it's conceptually a different test than the one that's described in the test name but with the same setup. However, I've never had a problem with this style of unit tests in practice so it seems fine.